### PR TITLE
test: add unit test to verify group permission behavior

### PR DESCRIPTION
**Pull request by Emyrk**
Only running with in memory db as there is no advantage to using posgres, and it would just take longer.

### CODE REVIEW

File: coderd/database/dbauthz/groupsauth_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+			} else {
+				require.Error(t, err, "group read")
+			}
+
+			members, err := db.GetGroupMembersByGroupID(actorCtx, group.ID)
+			if tc.ReadMembers {
+				require.NoError(t, err, "member read")
+				require.Len(t, members, tc.MembersExpected, "member count found does not match")
+			} else {
+				require.Error(t, err, "member read")
```
Comment (johnstcn) [inline coderd/database/dbauthz/groupsauth_test.go:163]: Do we want to check `dbauthz.IsUnauthorizedError()` here?

File: coderd/database/dbauthz/groupsauth_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+				Scope:  rbac.ExpandableScope(rbac.ScopeAll),
+			},
+			ReadGroup:   false,
+			ReadMembers: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
```
Comment (johnstcn) [inline coderd/database/dbauthz/groupsauth_test.go:147]: fun idea: you could actually just generate the name here from `${tc.Subject.ID}_ReadGroup${tc.ReadGroup}_ReadMembers${tc.ReadMembers}`

File: coderd/database/dbauthz/groupsauth_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	}), coderdtest.AccessControlStorePointer())
+
+	ownerCtx := dbauthz.As(context.Background(), rbac.Subject{
+		ID:     "owner",
+		Roles:  rbac.Roles(must(rbac.RoleIdentifiers{rbac.RoleOwner()}.Expand())),
+		Groups: []string{},
+		Scope:  rbac.ExpandableScope(rbac.ScopeAll),
+	})
+
+	org := dbgen.Organization(t, db, database.Organization{})
```
Comment (johnstcn) [inline coderd/database/dbauthz/groupsauth_test.go:42]: should we test multi-org here?

**Review by johnstcn — APPROVED** [review]

File: coderd/database/dbauthz/groupsauth_test.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	db = dbauthz.New(db, authz, slogtest.Make(t, &slogtest.Options{
+		IgnoreErrors: true,
+	}), coderdtest.AccessControlStorePointer())
+
+	ownerCtx := dbauthz.As(context.Background(), rbac.Subject{
+		ID:     "owner",
+		Roles:  rbac.Roles(must(rbac.RoleIdentifiers{rbac.RoleOwner()}.Expand())),
+		Groups: []string{},
+		Scope:  rbac.ExpandableScope(rbac.ScopeAll),
+	})
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply coderd/database/dbauthz/groupsauth_test.go:39]: Yea, I needed the original owner role for setup. Given the pattern of the tabled tests, it felt weird to not fully define the role in the table again

File: coderd/database/dbauthz/groupsauth_test.go (hunk 3 above)
Comment (Emyrk) [reply coderd/database/dbauthz/groupsauth_test.go:42]: https://github.com/coder/coder/blob/80bbcac4e8463445df6802c40fe5a59e875001dd/coderd/database/dbauthz/groupsauth_test.go#L136-L136

I threw in a different org admin. We don't actually need another org in the db for the authz test.

**Pull request merged**